lib/core/src/Cardano/Wallet.hs

```diff
@@ -2842,10 +2835,8 @@ data WalletFollowLog
 
 -- | Log messages from API server actions running in a wallet worker context.
 data WalletLog
-    = MsgSelectionStart UTxOIndex (NonEmpty TxOut)
-    | MsgSelectionError SelectionError
-    | MsgSelectionReportSummarized SelectionReportSummarized
-    | MsgSelectionReportDetailed SelectionReportDetailed
+    = MsgSelectAssets UTxOIndex TransactionCtx [TxOut] (BracketLog' (Either ErrSelectAssets (SelectionResult TokenBundle)))
```

Comment:
My comment from the previous review applies here:

https://github.com/input-output-hk/cardano-wallet/pull/2877#discussion_r705265581

Reply:
For normal data types it is reasonable to use record fields, but these types are just for logging. The message constructors are always used in the vicinity of the type definition. We want the definitions to be compact and unobtrusive, and we want it to be easy to add new log message constructors.

Reply:
In general, I agree with Jonathan here. I have no idea how to produce and consume a value of this type, and I have no helpers or commentary to guide me, and adding helpers or commentary is hindered by the fact that it's all embedded. But if you truly feel the benefits outweigh the costs, and that this will only be used in a limited context, then I don't mind*.

*until I have to touch it